Test failure on main only

https://github.com/py-cov-action/python-coverage-comment-action/actions/runs/14822239250/job/41610773397?pr=535

```
>                   warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
E                   pytest.PytestUnraisableExceptionWarning: Exception ignored in: <ssl.SSLSocket fd=-1, family=2, type=1, proto=6>
E                   
E                   Traceback (most recent call last):
E                     File "/usr/lib/python3.12/contextlib.py", line 548, in _push_exit_callback
E                       def _push_exit_callback(self, callback, is_sync=True):
E                       
E                   ResourceWarning: unclosed <ssl.SSLSocket fd=14, family=2, type=1, proto=6, laddr=('10.1.0.20', 36320), raddr=('185.199.108.153', 443)>

/home/runner/work/python-coverage-comment-action/python-coverage-comment-action/.venv/lib/python3.12/site-packages/_pytest/unraisableexception.py:85: PytestUnraisableExceptionWarning
---------------------------- Captured stdout setup -----------------------------
Deleting repository python-coverage-comment-action-end-to-end-private

---------- coverage: platform linux, python 3.12.3-final-0 -----------
Name                                Stmts   Miss Branch BrPart  Cover   Missing
-------------------------------------------------------------------------------
coverage_comment/__init__.py            0      0      0      0   100%
coverage_comment/__main__.py            6      0      2      0   100%
coverage_comment/activity.py           11      0      6      0   100%
coverage_comment/badge.py              31      0     10      0   100%
coverage_comment/comment_file.py        4      0      0      0   100%
coverage_comment/communication.py       8      0      0      0   100%
coverage_comment/coverage.py          126      0     18      0   100%
coverage_comment/diff_grouper.py       12      0      4      0   100%
coverage_comment/files.py              59      0      2      0   100%
coverage_comment/github.py            118      0     16      0   100%
coverage_comment/github_client.py      67      0     14      0   100%
coverage_comment/groups.py             35      0      6      0   100%
coverage_comment/log.py                 5      0      0      0   100%
coverage_comment/log_utils.py           9      0      0      0   100%
coverage_comment/main.py              131      0     20      0   100%
coverage_comment/settings.py          120      0     14      0   100%
coverage_comment/storage.py            69      0      6      0   100%
coverage_comment/subprocess.py         26      0      0      0   100%
coverage_comment/template.py          117      0     22      0   100%
-------------------------------------------------------------------------------
TOTAL                                 954      0    140      0   100%
Coverage HTML written to dir htmlcov

=========================== short test summary info ============================
ERROR tests/end_to_end/test_all.py::test_private_repo - pytest.PytestUnraisableExceptionWarning: Exception ignored in: <ssl.SSLSocket fd=-1, family=2, type=1, proto=6>

Traceback (most recent call last):
  File "/usr/lib/python3.12/contextlib.py", line 548, in _push_exit_callback
    def _push_exit_callback(self, callback, is_sync=True):
    
ResourceWarning: unclosed <ssl.SSLSocket fd=14, family=2, type=1, proto=6, laddr=('10.1.0.20', 36320), raddr=('185.199.108.153', 443)>
=================== 252 passed, 1 error in 190.15s (0:03:10) ===================
```

Given it's linked to pytest's unraisable hook, and it was improved in (unreleased) pytest 8.4, I'll try switching temporarily to the unreleased pytest.

--- a/coverage_comment/main.py
+++ b/coverage_comment/main.py
@@ -33,20 +33,22 @@ def main():
         log.info("Starting action")
         config = settings.Config.from_environ(environ=os.environ)
 
-        github_session = httpx.Client(
-            base_url=config.GITHUB_BASE_URL,
-            follow_redirects=True,
-            headers={"Authorization": f"token {config.GITHUB_TOKEN}"},
-        )
-        http_session = httpx.Client()
         git = subprocess.Git()
 
-        exit_code = action(
-            config=config,
-            github_session=github_session,
-            http_session=http_session,
-            git=git,
-        )
+        with (
+            httpx.Client(
+                base_url=config.GITHUB_BASE_URL,
+                follow_redirects=True,
+                headers={"Authorization": f"token {config.GITHUB_TOKEN}"},
+            ) as github_session,
+            httpx.Client() as http_session,
+        ):
+            exit_code = action(
+                config=config,
+                github_session=github_session,
+                http_session=http_session,
+                git=git,
+            )
 
         log.info("Ending action")
         sys.exit(exit_code)

--- a/tests/end_to_end/conftest.py
+++ b/tests/end_to_end/conftest.py
@@ -11,6 +11,7 @@ import shutil
 import subprocess
 import time
 
+import httpx
 import pytest
 import tenacity
 
@@ -387,3 +388,9 @@ def add_coverage_line(git, code_path):
         git("commit", "-m", "improve coverage")
 
     return f
+
+
+@pytest.fixture
+def http_client():
+    with httpx.Client() as client:
+        yield client

--- a/tests/end_to_end/test_all.py
+++ b/tests/end_to_end/test_all.py
@@ -2,7 +2,6 @@ from __future__ import annotations
 
 import base64
 
-import httpx
 import pytest
 
 
@@ -24,6 +23,7 @@ def test_public_repo(
     token_other,
     gh_create_fork,
     gh_other_username,
+    http_client,
 ):
     # Create a GitHub repo, make it public
     gh_create_repo("--public")
@@ -60,14 +60,12 @@ def test_public_repo(
     # - coverage branch readme url
     assert len(links) == 5
 
-    client = httpx.Client()
-
     # Check that all 5 links are valid and lead to a 200
     # It's made this way to avoid hardcoding links in the test, because I assume
     # they'll be evolving.
     number_of_svgs = 0
     for link in links:
-        response = client.get(link)
+        response = http_client.get(link)
         response.raise_for_status()
         number_of_svgs += int(response.text.startswith("<svg"))
 
@@ -81,7 +79,7 @@ def test_public_repo(
     raw_url_prefix = f"https://github.com/{repo_full_name}/raw/python-coverage-comment-action-data-my-great-project"
 
     readme_url = f"{raw_url_prefix}/README.md"
-    response = client.get(readme_url, follow_redirects=True)
+    response = http_client.get(readme_url, follow_redirects=True)
     response.raise_for_status()
     # And all previously found links should be present
     readme = response.text
@@ -91,13 +89,13 @@ def test_public_repo(
     # And while we're at it, there are 2 other files we want to check in this
     # branch. Once again, trying to avoid hardcoding too many specifics, that's what
     # unit tests are for.
-    data = client.get(f"{raw_url_prefix}/data.json", follow_redirects=True).json()
+    data = http_client.get(f"{raw_url_prefix}/data.json", follow_redirects=True).json()
     assert "coverage" in data
     assert "raw_data" in data
     assert "meta" in data["raw_data"]
     assert "coverage_path" in data
 
-    endpoint = client.get(
+    endpoint = http_client.get(
         f"{raw_url_prefix}/endpoint.json", follow_redirects=True
     ).json()
     assert "schemaVersion" in endpoint

